src/components/cities/CityEditForm.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useState } from 'react'
+import * as yup from 'yup'
+import { useRouter } from 'next/router'
+import { useMutation, UseQueryResult } from 'react-query'
+import { useTranslation } from 'next-i18next'
+import { AxiosError, AxiosResponse } from 'axios'
+import { Button, Grid, InputLabel, MenuItem, Select, Typography } from '@mui/material'
+
+import { routes } from 'common/routes'
+import { useEditCity } from 'service/city'
+import { AlertStore } from 'stores/AlertStore'
+import GenericForm from 'components/common/form/GenericForm'
+import SubmitButton from 'components/common/form/SubmitButton'
+import FormTextField from 'components/common/form/FormTextField'
+import { ApiErrors } from 'service/apiErrors'
+import { CityResponse, CityFormData, CityInput } from 'gql/cities'
+import { useCity, useCountriesList } from 'common/hooks/cities'
+import makeStyles from '@mui/styles/makeStyles'
+import createStyles from '@mui/styles/createStyles'
+
+const useStyles = makeStyles((theme) =>
+  createStyles({
+    heading: {
+      marginBottom: theme.spacing(5),
+      color: theme.palette.primary.dark,
+      textAlign: 'center',
+    },
+  }),
+)
+
+const validationSchema: yup.SchemaOf<CityFormData> = yup
+  .object()
+  .defined()
+  .shape({
+    name: yup.string().trim().min(3).max(25).required(),
+    postalCode: yup.string().trim().min(1).max(10).required(),
+    countryId: yup.string().trim().required(),
+  })
+
+type Props = {
+  id: string
+}
+export default function CityForm({ id }: Props) {
+  const classes = useStyles()
+  const countries = useCountriesList()
+  const { t } = useTranslation()
+  const router = useRouter()
+  const { data }: UseQueryResult<CityResponse> = useCity(id)
+  const [countryId, setCountryId] = useState(data?.countryId)
+
+  const initialValues: CityInput = {
+    name: data?.name,
+    postalCode: data?.postalCode,
+    countryId: countryId,
+  }
+
+  const mutation = useMutation<AxiosResponse<CityResponse>, AxiosError<ApiErrors>, CityInput>({
+    mutationFn: useEditCity(id),
+    onError: () => AlertStore.show(t('common:alerts.error'), 'error'),
+    onSuccess: () => {
+      AlertStore.show(t('common:alerts.message-sent'), 'success')
+      router.push(routes.admin.cities.home)
+    },
+  })
+
+  const onSubmit = (values: CityInput) => {
+    const data = {
+      name: values.name,
+      postalCode: values.postalCode,
+      countryId: countryId,
+    }
+    mutation.mutate(data)
+  }
+
+  function handleChange(event: any) {
```

Comment:
Please avoid using `any` as type

Here we most probably need the following type (not tested)

```suggestion
  function handleChange(event: SelectChangeEvent) {
```